### ISSUE TITLE
Unwanted request cancel after about two minutes

### ISSUE THREAD

**Issue by bartekfj** [cypress-io/cypress#2055]
### Is this a Feature or Bug?

Bug

### Current behavior:

I'm sending request to server that takes couple of minutes to get response.
Every single time right after two minutes request fails in Cypress.
Then Cypress sends the same request right after that which causes unwanted behavior in webapp.
In Cypress console log I can see two requests but in Chrome dev tools there is only one.
"responseTimeout" is set to 300000 (5 minutes).
![req_fail](https://user-images.githubusercontent.com/39901475/42026115-932c7de2-7ac6-11e8-996d-6227efef019f.png)

There is no issue when request is sent manually with the same version of Chrome and OS. 
![req_pass](https://user-images.githubusercontent.com/39901475/42026106-90bc3188-7ac6-11e8-99e3-41636e21e10a.png)

### Desired behavior:

Cypress should wait for response according to "responseTimeout" that is longer than two minutes.

### Steps to reproduce:

Send request that takes more than two minutes to get response.

### Versions

Cypress version: 3.0.1
Browser: Chrome 67
OS: Ubuntu Linux 16.04 LTS

**Comment by jennifer-shehane** [comment]
Hi, could you post the test code you are running? Feel free to blur our request urls if needed.

**Comment by bartekfj** [comment]
Hey,
I'm sorry for not providing enough information.
Request is generated by button click, so test is simulating user behavior.
Here is simplified version of my test.
```js
describe('create request', () => {

  before(() => {
    cy.login()
  })

  after(() => {
    cy.logout()
  })

  it('create request', () => {
    cy.visit(Cypress.env('url'))
    cy.contains('h1', 'Create request')
    cy.get('button').contains('launch').click()
    cy.get('#default').should('be.visible').click()
    cy.get('.page-form')
    cy.get('#input').clear().type('request name')
      .should('have.value', 'request name')
    cy.get('#option1').type('test')
    cy.get('#option2').click()
    // request fails on this step after two minutes
    cy.get('couple of selectors here').contains('Create').click()
    cy.get('.page-content', {timeout: 200000})
      .should('contain', 'Created successfully!')
  })

})
```

Also i was able to achieve similar behavior with [this example application](https://github.com/oleg-agapov/flask-vue-spa) with sleep set to 150 seconds on backend and "responseTimeout" set to 5 minutes in Cypress.
```python
# chunk of modified run.py file from example application
@app.route('/api/random')
def random_number():
    sleep(150)
    response = {
        'randomNumber': randint(1, 100)
    }
    return jsonify(response)
```

This test will fail every time right after 2 minutes.
```js
describe('experimental test', () => {

  it('timeout test', () => {
    cy.visit('http://127.0.0.1:8080')
  })

})
```

![fv_fail](https://user-images.githubusercontent.com/39901475/42080523-4523b6d4-7b83-11e8-8955-f73148f7684a.png)

**Comment by pokorson** [comment]
@jennifer-shehane does Cypress intercept somehow http requests coming from visited webpage? Is there any close after timeout or retry mechanism? If so can you point module where it might happen?

**Comment by bartekfj** [comment]
Some update from my side.
Node http server by design has [timeout](https://nodejs.org/api/http.html#http_server_timeout) set to 2 minutes.
Because Cypress proxies all traffic through this server, requests are failing right after 2 minutes.
I was able to fix this problem for modified example application case by [changing timeout](https://nodejs.org/api/http.html#http_server_settimeout_msecs_callback) in `Cypress/resources/app/packages/server/lib/server.js` file (Cypress CDN version).
```js
  // chunk of Cypress server.js file
  _this._server = http.createServer(app);
  _this._server.setTimeout(0); // added line
  allowDestroy(_this._server);
```
Problem still exist for webapp from my first comment that I'm working with. 
Why there is a second request sent right after 2 minutes which is visible in Cypress log is still not determined.

**Comment by jennifer-shehane** [comment]
Thanks for looking into this @bartekfj, we'll have to discuss exactly how this should be implemented for use within Cypress, but I believe this should be configurable as we've seen many issues of socket timeouts that may be negated by increasing this beyond the 2 minutes default.

**Comment by jennifer-shehane** [comment]
We will be closing this issue due to inactivity since no one has commented in a year and half. Please comment if there is new information to provide concerning the original issue and we can reopen.

**Comment by flotwig** [comment]
I can try to see if I can reproduce this, afaik we do not have code explicitly preventing this but it could still be an issue in 2020.

**Comment by martin-dev83** [comment]
I cannot post my code unfortunately here is the behavior I have:
I upload an excel with around 15k rows inside and doing some computations on it.

- the request goes to the web api (request is pending)

- **after two minutes**, I can see that my web api is receiving a second call, the same as the first one. **This second call is not sent by my frontend, nothing appears in the dev tools network tab and the code calling my web api is not triggered**

- when I manually stop the test I can see the second call to my api appearing in the cypress logs

Hope it can help, I am using Windows 10 - Cypress 4.11 - Node 10.18.1

**_Solution to my issue : Split the excel in multiple chunks taking less than 2 minutes to be uploaded_**

**Comment by flotwig** [comment]
Closing this issue since it has been fixed: #6426
